generic3g/actions/AccumulatorAction.F90

```diff
@@ -0,0 +1,302 @@
+#include "MAPL_Generic.h"
+module mapl3g_AccumulatorAction
+   use mapl3g_AccumulatorConstants
+   use mapl3g_ExtensionAction
+   use MAPL_ExceptionHandling
+   use ESMF
+   implicit none
+   private
+   public :: AccumulatorAction
+
+   type, extends(ExtensionAction) :: AccumulatorAction
+      integer(kind=c_int) :: accumulator_type
+      type(AccumulatorFunctions), pointer :: functions => null()
+      type(ESMF_Field), allocatable :: accumulation_field
+      type(ESMF_Field), allocatable :: counter_field
+   contains
+      procedure :: initialize => initialize_action
+      procedure :: update => update_action
+      procedure :: invalidate => invalidate_action
+      procedure, private :: has_coupler
+      procedure, private :: has_counter
+      procedure, private :: is_active
+   end type AccumulatorAction
+
+   interface AccumulatorAction
+      module procedure :: construct_accumulator_action
+   end interface AccumulatorAction
+
+   enum, bind(c)
+      enumerator :: SIMPLE_ACCUMULATOR
+      enumerator :: MEAN_ACCUMULATOR
+      enumerator :: MAX_ACCUMULATOR
+      enumerator :: MIN_ACCUMULATOR
+   end enum
+
+contains
+
+   function construct_accumulator_action(accumulator_type)
+      type(AccumulatorAction) :: acc
+      integer(kind=SIMPLE_ACCUMULATOR), intent(in) :: accumulator_type
+
+      acc%accumulator_type = accumulator_type
+
+   end function construct_accumulator_action
+
+   subroutine set_accumulator_action(acc) result(acc)
```

Comment:
We have a convention for the names for these.  (Might need to be added to style guide.)
```suggestion
   subroutine new_AccumulatoAction(acc) result(acc)
```